src/sbm.jl

```diff
@@ -1049,3 +1052,35 @@ function update_after_subsurfaceflow(sbm::SBM, zi, exfiltsatwater)
         sbm.zi[i] = zi[i]
     end
 end
+
+function update_total_water_storage(sbm::SBM, rnet, rrouting, lrouting)
+
+    # Set the total storage to zero
+    fill!(sbm.total_storage, 0)
+
+    # Burn the river routing values
+    sbm.total_storage[rnet] = (
+        rrouting.h_av .* sbm.riverfrac[rnet] * 0.001 # convert to mm
+    )
+
+    # Chunk the data for parallel computing
+    threaded_foreach(1:sbm.n, basesize=1000) do i
+
+        # Cumulate per vertical type
+        # Maybe re-categorize in the future
+        # Unique are those that are not always present
+        unique = sbm.glacierstore[i] * sbm.glacierfrac[i]
```

Comment:
Actually snow is also not always present, you might as well just add these variables to the surface part.